sys/include/net/nanocoap.h

```diff
@@ -347,6 +347,21 @@ void *coap_request_ctx_get_context(const coap_request_ctx_t *ctx);
  */
 uint32_t coap_request_ctx_get_tl_type(const coap_request_ctx_t *ctx);
 
+#if defined(MODULE_SOCK_UDP) || DOXYGEN
```

Comment:
We could have `coap_request_ctx_get_remote_udp()` declared unconditionally, in the manner of `coap_request_ctx_get_tl_type()`. If the inclusion of `#include "net/sock/udp.h"` is a problem, we could write it like this:
```C
#if defined(MODULE_SOCK_UDP)
const sock_udp_ep_t *
#else
const void *
#endif
coap_request_ctx_get_remote_udp(const coap_request_ctx_t *ctx);
```
, but I think this is less clean then simply including `net/sock/udp.h`.

Reply:
> I think this is less clean than simply including `net/sock/udp.h`

We can't include that unconditionally as it will fail if no sock implementation is present.

Reply:
Ah ok, because of `#include "sock_types.h"`, I see. Theoretically, this is solvable if we wrap the `#include "sock_types.h"` statement with an `#if` for gnrc, lwip and openwsn, instead of  having to think about whether there is a socket implementation.  Would this be an improvement in your opinion?

Reply:
How about

```C
#if !defined(MODULE_SOCK_UDP)
typedef void *sock_udp_ep_t;
endif
```

Reply:
```C
#if !defined(MODULE_SOCK_UDP)
typedef void sock_udp_ep_t;
#else
#include "net/sock/udp.h"
#endif
```
Yes, this is a good solution as well.